### DISCUSSION TITLE
Requests to backend service blocked when coming from frontend application

### DISCUSSION THREAD

**Discussion by otherippo** [authelia/authelia#4497] [Q&A] [answered]
### Version

v4.37.2

### Deployment Method

Docker

### Reverse Proxy

SWAG

### Description

Hi, I'm not sure this is a bug or I'm missing something in the configuration. 

I'm running Authelia with swag as a docker-compose configuration. Protected endpoints are accessible after login on authelia's login form, and when I access a frontend application (made with vue) I first also need to login - so far so good.

So I can access everything after login. But when the frontend application tries to access some resource from a backend service, the request it sends is redirected to Authelia and I get this response:

```
<!DOCTYPE html>
<html lang="en">
<head>
  <base href="https://rtc.cam.vader.in/authelia/" />
  <meta property="csp-nonce" content="J7SNQmhn1hSNwN8a0DWrHu44UyckgqLj" />
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="theme-color" content="#000000" />
  <meta name="description" content="Authelia login portal for your apps" />
  <link rel="manifest" href="./manifest.json" />
  <link rel="icon" href="./favicon.ico" />
  <title>Login - Authelia</title>
  <script type="module" crossorigin src="./static/js/index.81d2a771.js"></script>
  <link rel="stylesheet" href="./static/css/index.40feef90.css">
</head>

<body
    data-basepath="/authelia"
    data-duoselfenrollment="false"
    data-logooverride="false"
    data-rememberme="true"
    data-resetpassword="true"
    data-resetpasswordcustomurl=""
    data-theme="dark"
>
  <noscript>You need to enable JavaScript to run this app.</noscript>
  <div id="root"></div>
  
</body>
</html>
```

Obviously this could be fixed by not protecting this backend service by Authelia. But this is not a solution since it's open to the public by its own subdomain and needs protection. What am I missing here?

### Reproduction

A protected frontend application makes a request to a protected backend service. The request gets redirected to Authelia which returns the above HTML page saying `You need to enable JavaScript to run this app.`

### Logs (Authelia)

```shell
level=debug msg="Check authorization of subject username= groups= ip=xxx.xxx.xxx.xxx and object https://api.example.com/api/streams (method )."

level=debug msg="No matching rule for subject username= groups= ip=xxx.xxx.xxx.xxx and url https://api.example.com/api/streams (method ) applying default policy"

level=info msg="Access to https://api.example.com/api/streams (method unknown) is not authorized to user <anonymous>, responding with status code 401" method=GET path=/api/verify remote_ip=xxx.xxx.xxx.xxx
```

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)
- [ ] This is a bug report and not a support request
- [X] I have read the security policy and this bug report is not a security issue or security related issue
- [ ] I have either included the complete configuration file or I am sure it's unrelated to the configuration
- [X] I have provided all of the required information in full with the only alteration being reasonable sanitization in accordance with the [Troubleshooting Sanitization](https://www.authelia.com/r/sanitize) reference guide
- [X] I have checked for related proxy or application logs and included them if available
- [X] I have checked for related issues and checked the documentation

**Comment by james-d-elliott** [comment]
This is probably not a bug, and without specific information to confirm it is the case we ask people use discussions to make our workflow easier.

All requests routed via the proxy to the application will be protected if you decide to protect the application. You have a few options to handle this use case.

1. Configure an [access control rule](https://www.authelia.com/configuration/security/access-control/) targeting the backend application / applications [network / networks](https://www.authelia.com/configuration/security/access-control/#networks) as exceptions and either:
    1. Set the [policy](https://www.authelia.com/configuration/security/access-control/#policy)to one_factor and create a username and password for automated applications and include this information in the `Proxy-Authorization` header encoded in the basic scheme.
    2. Set the [policy](https://www.authelia.com/configuration/security/access-control/#policy) to bypass.
2. Ensure the requests from the backend applications are not done via the proxy which can be done using:
    1. The `container_name` option as the DNS name if you're using docker compose i.e. `http://<container_name>`. Several of our examples configure containers with the `container_name` attribute, where authelia is accessible via the `container_name` of `authelia` making its URL `http://authelia:9091`.
    2. An IP and port combination such as `http://127.0.0.1:80`.

**Comment by otherippo** [comment]
Okay makes sense and thank you for the quick answer!

Addressing the options you suggested:

1. access control rule
  i. Where would I add this header (e.g. in which location section) in the nginx config? Or does the frontend need to send this (similar to an authentication token / session cookie)?
  ii. But this would make the backend unprotected, right? Say when a user directly accesses it he doesn't need to login first?
2. requests from backend
  Do you mean the response the backend sends back? This is not an issue I think, since if I can reach it I also get its response. Also I already use the container names to route traffic (like `proxy_pass http://backend:8080/` inside a location block), and backend applications accessing each other are not done via the proxy.

Basically what I'd like to achieve is:

- User directly accessing the API (e.g. the backend application)
- User accessing the frontend application
- Frontend application accessing the API

should all be protected, so none of them can be accessed by a non authenticated user. The last part (last bullet point) is usually done by the frontend sending the session cookie (or token it got from the API) in a header, but I don't know if this is possible with Authelia

**Comment by otherippo** [comment, marked as answer]
Today I was able to solve the problem. As axios is used in the Vue frontend I just had to include `withCredentials: true`. This means Authelia's session cookie (if present) gets sent in the request's header: So if a user using the frontend application is logged in, its requests to the backend are already authenticated and everything works as expected.
